### PR TITLE
feat: UI sounds

### PR DESCRIPTION
**Pull request by fcolarich**
Hi hello!
This PR addresses a couple of things, mostly UI audio, but also a generic way to create "audio clip configs" which can contain more than one audio clip and will allow to setup more complex behaviours like looping, playing random clips, starting at random positions in the clips, etc.
BUT that's not ready yet for this PR as all that is NOT needed for UI.
Each AudioClipConfig is associated with an AudioCategory which will define which AudioMixer will correspond to it. That allows us to handle volume seamlessly.

So, the basic idea is, we have a singleton UIAudioEventBus that allows us to send audio events from anywhere. This simplifies how we setup views and whatnot, and we only need to add a reference to the actual AudioClipConfig we wanna play and call the singleton to reproduce it in a shared AudioSource using PlayOneShot. We need 3 different AudioSources for UI as we have Music, UI and some Chat sounds that we need to discriminate with settings later.
If its a music track it is reproduced as a normal track (so we change the clip and call Play) but we do some fading before and after so its more pretty.

Here we also add the correct AudioMixer to the elements created by the SDK so they go to the World category and we can properly set the volume.

I have already added audio to many elements, but there are many that do not have a prefab so they require manual adding of components, usually a ButtonView so we can have proper button on click audio.

And that's basically all :)

BTW: I added a lot of audio files here, once we have all audios configured we can clean it up.

Where do we have audio already?
Right now I added the audio refs to all generic UI components & in most places of the Backpack :) I had in Chat and other places but as they dont used prefabs and where changed, I have to re-do it xD

### CODE REVIEW

File: Explorer/Assets/DCL/Character/CharacterMotion/Animation/ApplyAnimationMovementBlend.cs [hunk 1]
```diff
@@ -55,5 +55,16 @@ private static int GetMovementBlendId(Vector3 velocity, MovementKind speedState)
...
+        public static int GetMovementBlendId(MovementKind speedState)
+        {
+            return speedState switch
+                   {
+                       MovementKind.Walk => 1,
+                       MovementKind.Jog => 2,
+                       MovementKind.Run => 3,
+                       _ => 0,
+                   };
+        }
```
Comment (Kinerius) [inline Explorer/Assets/DCL/Character/CharacterMotion/Animation/ApplyAnimationMovementBlend.cs:68]: What's the purpose of this method? its duplicates another one

**Review by Kinerius — APPROVED** [review]

File: Explorer/Assets/DCL/Character/CharacterMotion/Animation/ApplyAnimationMovementBlend.cs (hunk 1 above)
Comment (fcolarich) [reply Explorer/Assets/DCL/Character/CharacterMotion/Animation/ApplyAnimationMovementBlend.cs:68]: the problem is that the other one required a Vector3 and if the vector was 0 it returned 0 xD
but this is not in use in this branch anyway, so I can remove it!

Comment (fcolarich) [reply Explorer/Assets/DCL/Character/CharacterMotion/Animation/ApplyAnimationMovementBlend.cs]: okay Removed that :)

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]

File: Explorer/Assets/DCL/Audio/Scripts/AudioPlaybackUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using UnityEngine;
+
+namespace DCL.Audio
```
[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/Audio/Scripts/AudioPlaybackUtilities.cs:3]: Thank you Misha for the review!
I actually moved the scripts to the base folder to keep with project structure as I see we do not use Scripts folder!

File: Explorer/Assets/DCL/PluginSystem/Global/UIAudioPlaybackPlugin.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            this.assetsProvisioner = assetsProvisioner;
+        }
+
+        public void Dispose()
+        {
+            if (UIAudioPlaybackController != null) {UIAudioPlaybackController.Dispose();}
+        }
+
+        public void InjectToWorld(ref ArchSystemsWorldBuilder<Arch.Core.World> builder, in GlobalPluginArguments arguments) { }
```
[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/PluginSystem/Global/UIAudioPlaybackPlugin.cs:27]: Yeah, in this context it still doesnt have any system, the original idea was for this plugin to handle all AudioPlayback, but as its functionality was stripped it ended up just instantiating the container. 
I will change it and might add a new plugin just for landscape sounds to make it clearer! 👍

File: Explorer/Assets/DCL/PluginSystem/Global/UIAudioPlaybackPlugin.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        public async UniTask InitializeAsync(AudioPluginSettings settings, CancellationToken ct)
+        {
+            UIAudioPlaybackController = (await assetsProvisioner.ProvideInstanceAsync(settings.UIAudioPlaybackControllerReference, ct: ct)).Value;
+            UIAudioPlaybackController.Initialize();
+        }
+
+        public class AudioPluginSettings : IDCLPluginSettings
+        {
+            [field: Header(nameof(UIAudioPlaybackPlugin) + "." + nameof(AudioPluginSettings))]
```
[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/PluginSystem/Global/UIAudioPlaybackPlugin.cs:37]: Nice!

File: Explorer/Assets/DCL/UI/PageSelector/PageSelectorView.cs [hunk 5]
```diff
@@ -13,5 +14,35 @@ public class PageSelectorView : MonoBehaviour
 
         [field: SerializeField]
         public Transform PagesContainer { get; private set; }
+
+
+        [Header("Audio")]
+        [field: SerializeField]
```
[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/UI/PageSelector/PageSelectorView.cs:20]: Thank you for pointing it out!! I didnt convert these to properties as I should have 🤦

File: Explorer/Assets/DCL/Audio/Scripts/UIAudioPlaybackController.cs [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+            if (!settings.AudioEnabled) return;
+
+            PlaySingleAudio(audioClipConfig);
+        }
+
+        private bool CheckAudioCategory(AudioCategory category)
+        {
+            if (category is not (AudioCategory.Chat or AudioCategory.Music or AudioCategory.UI))
+            { //We can only play UI sounds through this bus. Other sounds are discarded -> maybe add a log here?
+                return false;
```
[the remark this answers is not in the document]
Comment (fcolarich) [reply Explorer/Assets/DCL/Audio/Scripts/UIAudioPlaybackController.cs]: 🤦 thank you for spotting it!

**Review by mikhail-dcl — APPROVED** [review]
Thanks 💚

**Pull request merged**